lib/facter/framework/config/config_reader.rb

```diff
@@ -2,6 +2,8 @@
 
 module Facter
   class ConfigReader
+    @log = Log.new(self)
```

Comment:
Is the common practice in Facter to initialize the logger this early? why not add it in the `init` method, to be initialized only when it is used?

Reply:
`refresh_config` method is called 4 times, that means a new logger every time that method is called
```
[2021-01-20 17:08:33.766582 ] WARN Facter::ConfigReader - Facter failed to read config file /Users/oana.tanasoiu/projects/bad_file.conf with the following error: /Users/oana.tanasoiu/projects/bad_file.conf: 2: Key 'some invalid info' may not be followed by token: end of file
[2021-01-20 17:08:33.766905 ] WARN Facter::ConfigReader - Facter failed to read config file /Users/oana.tanasoiu/projects/bad_file.conf with the following error: /Users/oana.tanasoiu/projects/bad_file.conf: 2: Key 'some invalid info' may not be followed by token: end of file
[2021-01-20 17:08:33.772585 ] WARN Facter::ConfigReader - Facter failed to read config file /Users/oana.tanasoiu/projects/bad_file.conf with the following error: /Users/oana.tanasoiu/projects/bad_file.conf: 2: Key 'some invalid info' may not be followed by token: end of file
[2021-01-20 17:08:33.772945 ] WARN Facter::ConfigReader - Facter failed to read config file /Users/oana.tanasoiu/projects/bad_file.conf with the following error: /Users/oana.tanasoiu/projects/bad_file.conf: 2: Key 'some invalid info' may not be followed by token: end of file
```

Reply:
You  can memoize the logger
```ruby
@log ||= Log.new(self)
```